lib/puppet/provider/ssl_pkey/openssl.rb

```diff
@@ -28,6 +28,7 @@ def self.to_pem(resource, key)
   end
 
   def exists?
+    return false if resource[:regenerate]
```

Comment:
I think this `regenerate` parameter is a bad idea as it is inherently incompatible with idempotent resources.

Can you think of another way of achieving this (minimal create date, some kind of hash validation, etc.) instead?